src/main/java/com/arpnetworking/commons/math/BigDecimalAccumulator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Dropbox
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arpnetworking.commons.math;
+
+import java.math.BigDecimal;
+
+/**
+ * Accumulator implementation using {@link BigDecimal}. This has no error.
+ * Although the precision of the result is still subject to constraints of
+ * representation as a {@code double}.
```

Comment:
I read this as meaning the addition operation has no per-operation error but the truncation operation itself will still result in an error. Is that correct?

e.g. the classic
```
jshell> new BigDecimal(0.1).add(new BigDecimal(0.2)).doubleValue()
$7 ==> 0.30000000000000004
```

Reply:
Yes, that's correct. BigDecimal will accumulate error free (e.g. very large very high precision); however, those values may not be representable as double.